docs/contributor/updateability.md

```diff
@@ -0,0 +1,349 @@
+# Upgrading MicroShift
+
+Following document is meant to be a contributor's introduction
+to the feature of MicroShift's updateability.
+
+## On-disk artifacts
+
+New on-disk artifacts were added:
+- `/var/lib/microshift-backups/`
+    - `health.json`
+    - backup directories
+- `/var/lib/microshift/version`
+
+### `/var/lib/microshift-backups/`
+
+New, additional directory for keeping health and backup related artifacts.
+For ostree systems it's a non-configurable directory where backups are created.
+For non-ostree systems it's a default (configurable) directory for backups.
```

Comment:
I didn't realize this was configurable. Why is it configurable?

Reply:
It made sense since it's a cli command. If this is manual process, we don't really care where user will store it.

```
Usage:
  microshift admin data [command]

Available Commands:
  backup      Backup MicroShift data

Flags:
  -h, --help             help for data
      --name string      Backup name (default "4.15__20230721_072807")
      --storage string   Directory with backups (default "/var/lib/microshift-backups")
```

I just noticed that we missed command to restore :smile:

Reply:
Do you think we should change the cli? From `microshift admin data {backup,restore}` to something more concise?

Reply:
Changing in https://github.com/openshift/microshift/pull/2108